### PR TITLE
Add average precision score

### PR DESCRIPTION
**Pull request by msluszniak**
Based on #123

### CODE REVIEW

File: lib/scholar/metrics.ex [hunk 1]
```diff
@@ -555,6 +555,125 @@ defmodule Scholar.Metrics do
...
+        f32[5]
+        [1.0, 1.0, 0.5, 0.5, 0.0]
+      >
+      iex> thresholds
+      #Nx.Tensor<
+        f32[4]
+        [0.10000000149011612, 0.3499999940395355, 0.4000000059604645, 0.800000011920929]
+      >
+  """
+  defn precision_recall_curve(y_true, probabilities_predicted, distinct_value_indices, weights) do
```
[the remark this answers is not in the document]
Comment (josevalim) [reply lib/scholar/metrics.ex:586]: I pushed a PR here to convert the existing functions: #125. If you agree, we can merge it and port to this PR.

[the remark this answers is not in the document]
Comment (msluszniak) [reply lib/scholar/metrics.ex]: Now it works

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: lib/scholar/metrics.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (msluszniak) [reply lib/scholar/metrics.ex]: Ops, I didn't see message about the PR, but I make changes, that do the same :')

[the remark this answers is not in the document]
Comment (josevalim) [reply lib/scholar/metrics.ex]: No worries, I already pushed my changes on top!

**Pull request merged**